src/hooks/useLedgerImport.ts

```diff
@@ -0,0 +1,227 @@
+import TransportBLE from '@ledgerhq/react-native-hw-transport-ble';
+import { useCallback, useEffect, useState } from 'react';
+import { DebugContext } from '@/logger/debugContext';
+import { logger, RainbowError } from '@/logger';
+import { Subscription } from '@ledgerhq/hw-transport';
+import { Alert } from 'react-native';
+import {
+  checkAndRequestAndroidBluetooth,
+  showBluetoothPermissionsAlert,
+  showBluetoothPoweredOffAlert,
+} from '@/utils/bluetoothPermissions';
+import { IS_ANDROID, IS_IOS } from '@/env';
+import { ledgerErrorHandler, LEDGER_ERROR_CODES } from '@/utils/ledger';
+
+enum LEDGER_IMPORT_STATUS {
+  SEARCHING = 'SEARCHING',
+  FOUND = 'FOUND',
+  PAIRING = 'PAIRING',
+  PAIRED = 'PAIRED',
+  ERROR = 'ERROR',
+}
+
+/**
+ * React hook used for checking ledger connections and handling connnection error states
+ */
+export function useLedgerImport({
+  errorCallback,
+  successCallback,
+}: {
+  successCallback?: (deviceId: string) => void;
+  errorCallback?: (errorType: LEDGER_ERROR_CODES) => void;
+}) {
+  const [pairingStatus, setPairingStatus] = useState<LEDGER_IMPORT_STATUS>(
+    LEDGER_IMPORT_STATUS.SEARCHING
+  );
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [observer, setObserverSubscription] = useState<Subscription | null>(
+    null
+  );
+  const [listener, setListenerSubscription] = useState<Subscription | null>(
+    null
+  );
```

Comment:
Would be great if you made sure these need to be `useState`, if these values are not refreshing any views and are purely logical I would love to see them as refs